test/parallel/test-fs-promises-writefile.js

```diff
@@ -7,20 +7,70 @@ const path = require('path');
 const tmpdir = require('../common/tmpdir');
 const assert = require('assert');
 const tmpDir = tmpdir.path;
+const { Readable } = require('stream');
 
 tmpdir.refresh();
 
 const dest = path.resolve(tmpDir, 'tmp.txt');
 const otherDest = path.resolve(tmpDir, 'tmp-2.txt');
 const buffer = Buffer.from('abc'.repeat(1000));
 const buffer2 = Buffer.from('xyz'.repeat(1000));
+const stream = Readable.from(['a', 'b', 'c']);
+const stream2 = Readable.from(['a', 'b', 'c']);
+const iterable = {
+  [Symbol.iterator]: function*() {
+    yield 'a';
+    yield 'b';
+    yield 'c';
+  }
+};
+const asyncIterable = {
+  async* [Symbol.asyncIterator]() {
+    yield 'a';
+    yield 'b';
+    yield 'c';
+  }
+};
 
 async function doWrite() {
   await fsPromises.writeFile(dest, buffer);
   const data = fs.readFileSync(dest);
   assert.deepStrictEqual(data, buffer);
 }
 
+async function doWriteStream() {
+  await fsPromises.writeFile(dest, stream);
+  let expected = '';
+  for await (const v of stream2) expected += v;
```

Comment:
I don't think it's correct to use async iteration here instead of specifying in advance what value you expect.
```suggestion
  const expected = 'abc';
```